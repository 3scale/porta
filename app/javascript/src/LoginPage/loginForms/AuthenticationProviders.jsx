// @flow

import * as React from 'react'
import { KeyIcon, LessThanIcon, GreaterThanIcon } from '@patternfly/react-icons'

export type ProvidersProps = {
  authorizeURL: string,
  humanKind: string
}

type Props = {
  authenticationProviders: Array<ProvidersProps>
}

const Provider = ({authorizeURL, humanKind}: {authorizeURL: string, humanKind: string}) => {
  return (
    <p className='login-provider'>
      <a className='login-provider-link' href={authorizeURL}>
        <KeyIcon/>{' Authenticate through '}
        <LessThanIcon/><GreaterThanIcon/>{humanKind}
      </a>
    </p>
  )
}

const AuthenticationProviders = (props: Props): React.Node => {
  const {authenticationProviders} = props
  const providersList = authenticationProviders.map(
    provider => <Provider authorizeURL={provider.authorizeURL} humanKind={provider.humanKind} key={provider.humanKind}/>
  )

  return (
    <React.Fragment>
      <div className='providers-list'>
        {providersList}
      </div>
    </React.Fragment>
  )
}

export {AuthenticationProviders}
