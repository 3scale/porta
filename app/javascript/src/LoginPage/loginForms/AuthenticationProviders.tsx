import { GreaterThanIcon, KeyIcon, LessThanIcon } from '@patternfly/react-icons'

import type { FunctionComponent } from 'react'

export type ProvidersProps = {
  authorizeURL: string,
  humanKind: string
}

type Props = {
  authenticationProviders: Array<ProvidersProps>
}

const Provider: FunctionComponent<ProvidersProps> = ({ authorizeURL, humanKind }) => (
  <p className='login-provider'>
    <a className='login-provider-link' href={authorizeURL}>
      <KeyIcon />{' Authenticate through '}
      <LessThanIcon /><GreaterThanIcon />{humanKind}
    </a>
  </p>
)

// eslint-disable-next-line react/no-multi-comp
const AuthenticationProviders: FunctionComponent<Props> = (props) => {
  const { authenticationProviders } = props
  const providersList = authenticationProviders.map(
    provider => <Provider key={provider.humanKind} authorizeURL={provider.authorizeURL} humanKind={provider.humanKind} />
  )

  return (
    <div className='providers-list'>
      {providersList}
    </div>
  )
}

export { AuthenticationProviders, Props }
