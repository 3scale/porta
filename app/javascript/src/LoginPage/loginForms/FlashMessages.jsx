// @flow

import * as React from 'react'
import { ExclamationCircleIcon } from '@patternfly/react-icons'
import type {FlashMessage} from 'Types'

const FlashMessages = ({flashMessages}: {flashMessages: Array<FlashMessage>}): React.Node => {
  const messagesList = flashMessages.map(message =>
    <p key={message.message}
      className={`pf-c-form__helper-text pf-m-${message.type}`}
    >
      <ExclamationCircleIcon />
      {` ${message.message}`}
    </p>
  )
  return (
    <div>
      {messagesList}
    </div>
  )
}

export {FlashMessages}
