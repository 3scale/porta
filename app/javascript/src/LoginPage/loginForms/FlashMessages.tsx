import { ExclamationCircleIcon } from '@patternfly/react-icons'
import type { FlashMessage } from 'Types'

const FlashMessages = (
  {
    flashMessages
  }: {
    flashMessages: Array<FlashMessage>
  }
): React.ReactElement => {
  const messagesList = flashMessages.map(message =>
    <p key={message.message}
      className={`pf-c-form__helper-text pf-m-${message.type}`}
    >
      <ExclamationCircleIcon />
      {` ${message.message}`}
    </p>
  )
  return (
    <div>
      {messagesList}
    </div>
  )
}

export { FlashMessages }
