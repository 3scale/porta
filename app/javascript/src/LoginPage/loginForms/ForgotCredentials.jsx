// @flow

import * as React from 'react'
import {LoginMainFooterBandItem} from '@patternfly/react-core'

const ForgotCredentials = ({requestPasswordResetPath}: {requestPasswordResetPath: string}): React.Node =>
  <LoginMainFooterBandItem>
    <a href={requestPasswordResetPath}>Forgot password?</a>
  </LoginMainFooterBandItem>

export {ForgotCredentials}
