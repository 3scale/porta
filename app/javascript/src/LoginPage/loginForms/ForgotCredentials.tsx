import * as React from 'react'
import { LoginMainFooterBandItem } from '@patternfly/react-core'

const ForgotCredentials = (
  {
    requestPasswordResetPath
  }: {
    requestPasswordResetPath: string
  }
): React.ReactElement => <LoginMainFooterBandItem>
  <a href={requestPasswordResetPath}>Forgot password?</a>
</LoginMainFooterBandItem>

export { ForgotCredentials }
