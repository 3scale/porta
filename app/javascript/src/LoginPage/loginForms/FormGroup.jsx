// @flow

import React from 'react'
import {
  FormGroup as PF4FormGroup,
  TextInput
} from '@patternfly/react-core'

type Props = {
  isRequired?: boolean,
  type: string,
  labelIsValid: boolean,
  inputProps: {
    value: 'string',
    onChange: () => void,
    autoFocus: 'string',
    inputIsValid: boolean,
    ariaInvalid: boolean
  }
}

const getFormGroupProps = (type: string) => {
  const types = ['username', 'password', 'email', 'user[username]', 'user[email]', 'user[first_name]', 'user[last_name]', 'user[password]', 'user[password_confirmation]']
  const typeIndex = types.indexOf(type)
  const inputTypes = ['text', 'password', 'email', 'text', 'email', 'text', 'text', 'password', 'password']
  const labels = ['Email or Username', 'Password', 'Email address', 'Username', 'Email', 'First name', 'Last name', 'Password', 'Password confirmation']
  const fieldIDs = ['session_username', 'session_password', 'email', 'user_username', 'user_email', 'user_first_name', 'user_last_name', 'user_password', 'user_password_confirmation']
  const helperTextInvalids = ['Email or username is mandatory', 'Password is mandatory', 'A valid email address is mandatory', 'Username is mandatory', 'A valid email is mandatory', '', '', 'Password is mandatory', 'Password confirmation is mandatory']
  const tabIndexs = ['1', '2', '1', null, null, null, null, null, null]
  return {
    label: labels[typeIndex],
    inputType: inputTypes[typeIndex],
    fieldId: fieldIDs[typeIndex],
    helperTextInvalid: helperTextInvalids[typeIndex],
    tabIndex: tabIndexs[typeIndex]
  }
}

const FormGroup = ({isRequired = false, type, labelIsValid, inputProps}: Props) => {
  const {value, onChange, autoFocus, inputIsValid, ariaInvalid} = inputProps
  const {label, inputType, fieldId, helperTextInvalid, tabIndex} = getFormGroupProps(type)
  return (
    <React.Fragment>
      <PF4FormGroup
        label={label}
        isRequired={isRequired}
        fieldId={fieldId}
        helperTextInvalid={helperTextInvalid}
        isValid={labelIsValid}
      >
        <TextInput
          isRequired={isRequired}
          type={inputType}
          id={fieldId}
          name={type}
          tabIndex={tabIndex}
          value={value}
          onChange={onChange}
          autoFocus={autoFocus}
          isValid={inputIsValid}
          aria-invalid={ariaInvalid}
        />
      </PF4FormGroup>
    </React.Fragment>
  )
}

export {FormGroup}
