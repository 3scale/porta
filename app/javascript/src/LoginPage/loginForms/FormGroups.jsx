// @flow

import * as React from 'react'
import {
  FormGroup as PF4FormGroup,
  TextInput
} from '@patternfly/react-core'
import type { InputProps } from 'Types'

type Props = {
  type: string,
  helperTextInvalid: string,
  inputProps: InputProps
}

const helperTexts = {
  emailOrUsername: 'Email or username is mandatory',
  username: 'Username is mandatory',
  email: 'A valid email address is mandatory',
  firstName: '',
  lastName: '',
  password: 'Password is mandatory',
  passwordConfirmation: {
    isMandatory: 'Password confirmation is mandatory',
    mustMatch: 'Password and Password confirmation must match'
  }
}

const FormGroup = ({type, helperTextInvalid, inputProps}: Props) => {
  const {isRequired, label, fieldId, isValid, name, value, onChange, onBlur, autoFocus, ariaInvalid} = inputProps
  return (
    <React.Fragment>
      <PF4FormGroup
        isRequired={isRequired}
        label={label}
        fieldId={fieldId}
        helperTextInvalid={helperTextInvalid}
        isValid={isValid}
      >
        <TextInput
          type={type}
          name={name}
          isRequired={isRequired}
          id={fieldId}
          value={value}
          onChange={onChange}
          onBlur={onBlur}
          autoFocus={autoFocus}
          isValid={isValid}
          aria-invalid={ariaInvalid}
        />
      </PF4FormGroup>
    </React.Fragment>
  )
}

const TextField = ({inputProps}: {inputProps: InputProps}): React.Node => {
  return (
    <FormGroup
      type='text'
      inputProps={inputProps}
      helperTextInvalid={helperTexts.emailOrUsername}
    />
  )
}

const EmailField = ({inputProps}: {inputProps: InputProps}): React.Node => {
  return (
    <FormGroup
      type='email'
      inputProps={inputProps}
      helperTextInvalid={helperTexts.email}
    />
  )
}

const PasswordField = ({inputProps}: {inputProps: InputProps}): React.Node => {
  let helperText = helperTexts.password
  return (
    <FormGroup
      type='password'
      inputProps={inputProps}
      helperTextInvalid={helperText}
    />
  )
}

const PasswordConfirmationField = ({inputProps}: {inputProps: InputProps}): React.Node => {
  const defaultErrorMessage = helperTexts.passwordConfirmation.isMandatory
  const errorMessage = inputProps.errorMessage
  let helperText = errorMessage ? helperTexts.passwordConfirmation[errorMessage] : defaultErrorMessage
  return (
    <FormGroup
      type='password'
      inputProps={inputProps}
      helperTextInvalid={helperText}
    />
  )
}

export { TextField, EmailField, PasswordField, PasswordConfirmationField }
