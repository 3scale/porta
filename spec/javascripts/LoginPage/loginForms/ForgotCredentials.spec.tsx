import { mount } from 'enzyme'
import { ForgotCredentials } from 'LoginPage/loginForms/ForgotCredentials'

import type { Props } from 'LoginPage/loginForms/ForgotCredentials'

const props: Props = {
  requestPasswordResetPath: 'login-path'
}

it('should render itself', () => {
  const wrapper = mount(<ForgotCredentials {...props} />)
  expect(wrapper).toMatchSnapshot()
})
