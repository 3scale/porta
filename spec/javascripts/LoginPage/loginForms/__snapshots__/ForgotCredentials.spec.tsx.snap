// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render itself 1`] = `
<ForgotCredentials
  requestPasswordResetPath="login-path"
>
  <LoginMainFooterBandItem>
    <p
      className="pf-c-login__main-footer-band-item"
    >
      <a
        href="login-path"
      >
        Forgot password?
      </a>
    </p>
  </LoginMainFooterBandItem>
</ForgotCredentials>
`;
